extensions/amp-story-shopping/0.1/amp-story-shopping-tag.js

```diff
@@ -70,6 +77,17 @@ export class AmpStoryShoppingTag extends AMP.BaseElement {
     );
   }
 
+  /**
+   * @private
+   */
+  onClick_() {
+    this.storeService_.dispatch(Action.ADD_SHOPPING_DATA, {
+      'activeProduct': this.storeService_.get(StateProperty.SHOPPING_DATA)[
+        this.element.getAttribute('data-tag-id')
+      ]['product-tag-id'],
+    });
+  }
```

Comment:
We can remove this round-trip here by passing the product data to the onClick, like:
```
/** @private */
onClick_(tagData) {
  this.storeService_.dispatch(Action.ADD_SHOPPING_DATA, {
    'activeProduct': tagData,
  });
}
```

Notice also how the object is being set as activeProduct instead of the tag id. 
This is beneficial because we wont have to look up the product in the map, we can just access it directly.
Either way works but setting it as the whole entry will be less code.